cpp/src/parquet/types.h

```diff
@@ -646,6 +650,86 @@ using DoubleType = PhysicalType<Type::DOUBLE>;
 using ByteArrayType = PhysicalType<Type::BYTE_ARRAY>;
 using FLBAType = PhysicalType<Type::FIXED_LEN_BYTE_ARRAY>;
 
+// Type forward declarations for TypeClasses
+class Comparator;
+class Statistics;
+
+template <typename DType>
+class TypedComparator;
+
+template <typename DType>
+class TypedStatistics;
+
+using BoolComparator = TypedComparator<BooleanType>;
+using Int32Comparator = TypedComparator<Int32Type>;
+using Int64Comparator = TypedComparator<Int64Type>;
+using Int96Comparator = TypedComparator<Int96Type>;
+using FloatComparator = TypedComparator<FloatType>;
+using DoubleComparator = TypedComparator<DoubleType>;
+using FLBAComparator = TypedComparator<FLBAType>;
+using ByteArrayComparator = TypedComparator<ByteArrayType>;
+
+using BoolStatistics = TypedStatistics<BooleanType>;
+using Int32Statistics = TypedStatistics<Int32Type>;
+using Int64Statistics = TypedStatistics<Int64Type>;
+using FloatStatistics = TypedStatistics<FloatType>;
+using DoubleStatistics = TypedStatistics<DoubleType>;
+using FLBAStatistics = TypedStatistics<FLBAType>;
+using ByteArrayStatistics = TypedStatistics<ByteArrayType>;
+
+template <typename DType>
+struct TypeClasses {};
```

Comment:
I intended this to provide mappings to other classes, too. I'm not sure it's needed at all now since I removed some of the subclasses that made me add this in the first place. Will take a look

Reply:
I completely removed this type traits, it was only needed in an intermediate state of this patch